plugins/inputs/docker_log/docker_logs.go

```diff
@@ -0,0 +1,472 @@
+	return "Plugin to get docker logs"
+}
+
+func (d *DockerLogs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *DockerLogs) Gather(acc telegraf.Accumulator) error {
+	/*Check to see if any new containers have been created since last time*/
+	return d.containerListUpdate(acc)
+}
+
+/*Following few functions have been inherited from telegraf docker input plugin*/
+func (d *DockerLogs) createContainerFilters() error {
+	filter, err := filter.NewIncludeExcludeFilter(d.ContainerInclude, d.ContainerExclude)
+	if err != nil {
+		return err
+	}
+	d.containerFilter = filter
+	return nil
+}
+
+func (d *DockerLogs) createLabelFilters() error {
+	filter, err := filter.NewIncludeExcludeFilter(d.LabelInclude, d.LabelExclude)
+	if err != nil {
+		return err
+	}
+	d.labelFilter = filter
+	return nil
+}
+
+func (d *DockerLogs) createContainerStateFilters() error {
+	if len(d.ContainerStateInclude) == 0 && len(d.ContainerStateExclude) == 0 {
+		d.ContainerStateInclude = []string{"running"}
+	}
+	filter, err := filter.NewIncludeExcludeFilter(d.ContainerStateInclude, d.ContainerStateExclude)
+	if err != nil {
+		return err
+	}
+	d.stateFilter = filter
+	return nil
+}
+
+func (d *DockerLogs) addToContainerList(containerId string, logReader io.ReadCloser) error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	d.containerList[containerId] = logReader
+	return nil
+}
+
+func (d *DockerLogs) removeFromContainerList(containerId string) error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	delete(d.containerList, containerId)
+	return nil
+}
+
+func (d *DockerLogs) containerInContainerList(containerId string) bool {
+	if _, ok := d.containerList[containerId]; ok {
+		return true
+	}
+	return false
+}
+
+func (d *DockerLogs) stopAllReaders() error {
+	for _, container := range d.containerList {
+		container.Close()
+	}
+	return nil
+}
+
+func (d *DockerLogs) containerListUpdate(acc telegraf.Accumulator) error {
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout.Duration)
+	defer cancel()
+	if d.client == nil {
+		return errors.New(fmt.Sprintf("%s : Dock client is null", ERR_PREFIX))
+	}
+	containers, err := d.client.ContainerList(ctx, d.opts)
+	if err != nil {
+		return err
+	}
+	for _, container := range containers {
+		if d.containerInContainerList(container.ID) {
+			continue
+		}
+		d.wg.Add(1)
+		/*Start a new goroutine for every new container that has logs to collect*/
+		go func(c types.Container) {
+			defer d.wg.Done()
+			logOptions := types.ContainerLogsOptions{
+				ShowStdout: true,
+				ShowStderr: true,
+				Timestamps: false,
+				Details:    true,
+				Follow:     true,
+				Tail:       "0",
+			}
+			logReader, err := d.client.ContainerLogs(context.Background(), c.ID, logOptions)
+			if err != nil {
+				acc.AddError(err)
+				return
+			}
+			d.addToContainerList(c.ID, logReader)
+			err = d.tailContainerLogs(c, logReader, acc)
+			if err != nil {
+				acc.AddError(fmt.Errorf("%s : %s ", ERR_PREFIX, err.Error()))
```

Comment:
Just do `acc.AddError(err)`, the accumulator will add the prefix for you.

Reply:
Done!